docs/observability/metric-naming-conventions.md

```diff
@@ -0,0 +1,355 @@
+backstage.plugin.catalog.operations.duration
+backstage.framework.database.operations.duration
+backstage.plugin.scaffolder.tasks.duration
+
+# Size measurements
+backstage.framework.httpRouter.request.size
+backstage.framework.database.result.size
+backstage.plugin.techdocs.document.size
+```
+
+### Gauges (current state)
+
+Gauges typically don't need special suffixes, but use descriptive names:
+
+```yaml
+backstage.framework.database.connections.count
+backstage.plugin.catalog.entities.count
+backstage.plugin.scaffolder.tasks.queued
+```
+
+## Attribute Naming
+
+Attributes (labels/tags) provide dimensional context to metrics. Use attributes to distinguish between different operations, statuses, and outcomes instead of creating separate metrics.
+
+### Standard Attributes
+
+**Always use these attributes when applicable:**
+
+```yaml
+# Operation and status context
+operation: "create" | "read" | "update" | "delete" | "refresh" | "validation"
+status: "success" | "error" | "timeout"
+
+# Entity context (for catalog metrics)
+entity.kind: "Component" | "API" | "Resource" | "System" | "Domain" | "..."
+entity.type: specific entity type
+entity.namespace: entity namespace
+
+# Error context (when status="error")
+error.type: error class name or type
+error.code: specific error code
+```
+
+### Attribute Naming Rules
+
+1. **Use `snake_case`** for attribute names
+2. **Be consistent** across related metrics
+3. **Keep cardinality reasonable** (avoid unique identifiers)
+4. **Use semantic names** that clearly indicate the attribute's purpose
+5. **Use attributes instead of separate metrics** for different statuses/operations
+
+```yaml
+# Good attributes
+entity.kind: 'Component'
+operation: 'refresh'
+status: 'success'
+db.table.name: 'entities'
+
+# Avoid high cardinality
+user.id: 'user123456' # Too many unique values
+request.id: 'req-abc-123' # Unique per request
+entity.name: 'my-service-api' # Too many unique values (use entity.kind instead)
+
+# Avoid unclear names
+type: 'thing' # Too generic
+t: 'Component' # Too abbreviated
+```
+
+## Plugin-Specific Examples
+
+### Catalog Plugin
+
+```yaml
+# Entity operations
+backstage.plugin.catalog.entities.operations.total
+# Attributes: { operation: "create" | "update" | "delete", entity.kind: "Component", status: "success" | "error" }
+
+# Processing operations
+backstage.plugin.catalog.operations.total
+# Attributes: { operation: "refresh" | "validation" | "provider_sync", status: "success" | "error" }
+
+backstage.plugin.catalog.operations.duration
+# Attributes: { operation: "refresh" | "validation", status: "success" | "error" }
+
+# Current state
+backstage.plugin.catalog.entities.count
+# Attributes: { entity.kind: "Component", entity.namespace: "default" }
+
+# Common attributes for all catalog metrics
+{
+  "entity.kind": "Component",
+  "entity.namespace": "default",
+  "operation": "refresh",
+  "provider": "github",
+  "status": "success"
+}
+```
+
+### Scaffolder Plugin
```

Comment:
A member of the OTEL community asked what we think about leveraging the existing [workflow semantic conventions](https://github.com/thompson-tomo/semantic-conventions/blob/feature/%231688_Workflows/docs/workflow/workflow-metrics.md) for the scaffolder metrics.

@benjdlambert, what are your thoughts on that idea?

Reply:
Hey sorry I missed this, from my brief look this looks like we should be doing this!